glusterfs.spec.in

```diff
@@ -53,6 +53,16 @@
 # rpmbuild -ta @PACKAGE_NAME@-@PACKAGE_VERSION@.tar.gz --with ipv6default
 %{?_with_ipv6default:%global _with_ipv6default --with-ipv6-default}
 
+# linux-io_uring
+# If you wish to compile an rpm without linux-io_uring support...
+# rpmbuild -ta  @PACKAGE_NAME@-@PACKAGE_VERSION@.tar.gz --disable-linux-io_uring
+%{?_without_linux_io_uring:%global _without_linux_io_uring --disable-linux-io_uring}
+
+# Disable linux-io_uring on unsupported distros.
+%if ( 0%{?fedora} && 0%{?fedora} <= 32 ) || ( 0%{?rhel} && 0%{?rhel} <= 7 )
+%global _without_linux_io_uring --disable-linux-io_uring
```

Comment:
Ok, so here you set `_without_linux_io_uring` on CentOS 7. You also use that when `./configure` is called further below.

I really do not think you need the hack in `configure.ac`.

Reply:
@nixpanic  I tried removing the configure.ac hack and also tried to add something like this
```
-%if ( 0%{?fedora} && 0%{?fedora} <= 32 ) || ( 0%{?rhel} && 0%{?rhel} <= 7 )
+%if ( 0%{?fedora} && 0%{?fedora} <= 32 ) || ( 0%{?rhel} && 0%{?rhel} <= 7 ) || ( 0%{?centos} && 0%{?centos} <= 7 )
```
But the compilation fails on CentOS7..

Reply:
That is strange, availability for `libtirpc` is similar. Just checking for `rhel` should be sufficient, no need to handle CentOS different.

Can you point to the build logs where this is failing? The `./configure` command should have the `--disable-linux-io_uring` option... Maybe something else went wrong somewhere?

Reply:
Removed the hack in `configure.ac`, it fails at https://build.gluster.org/job/gh_devrpm-el7/1256/console
In `glusterfs.spec.in` I have defined `without_linux_io_uring --disable-linux-io_uring` etc. Do I need to check this variable in configure.ac? Currently I check only `test "x$enable_linux_io_uring" != "xno"` in `configur.ac`

Reply:
I think that part just runs `make dist`, and that should not depend on any `--without-*` or `--disable-*` kind of flags. This is run before the RPM gets build. The RPM is built from the tarball that is generated with `make dist`.

There might be other cases where the distribution affects `make dist`. It is wrong, as everything should get included in the tarball, and during tarball creation there should be no linking against any libraries. If no fallback is possible (like libtirpc does), or really not wanted, you might need to add the `--disable-linux-io_uring` to the CI scripts.

Reply:
There is also a failure in the generated script:

    ./configure: line 15802: test: 4.8.5: integer expression expected

Not sure if it is introduced with this change though.

Reply:
> There is also a failure in the generated script:
> 
> ```
> ./configure: line 15802: test: 4.8.5: integer expression expected
> ```
> 
> Not sure if it is introduced with this change though.

Nope that is unrelated. Happens without this patch.